src/System.Buffers.ReaderWriter/System/Buffers/Reader/BufferReader_text.cs

```diff
@@ -27,9 +31,68 @@ public static partial class ReaderExtensions
         private static ParseDelegate<float> s_floatParse;
         private static ParseDelegate<double> s_doubleParse;
         private static ParseDelegate<decimal> s_decimalParse;
+        private static ParseDelegate<Guid> s_guidParse;
+        private static ParseDelegate<TimeSpan> s_timeSpanParse;
+        private static ParseDelegate<DateTime> s_dateTimeParse;
+        private static ParseDelegate<DateTimeOffset> s_dateTimeOffsetParse;
 
         // Note: We mark the getters as NoInlining as we want to make sure the delegate creation is lazy.
```

Comment:
Would it be better to do:
if(s_xxxParse != null) return s_xxxParse;
else return InitializeXxxParse();
... and then let the calls inline?

And more generally, do you think lazy creating all the delegates makes sense? We are paying non inlined call cost plus a branch, and it seems that at least some of the delegates will be quite commonly used.

Reply:
> Would it be better to do

I can try one and see if it measures.

> do you think lazy creating all the delegates makes sense?

As we have so many, I believe so. Everything is tuned so "normal" usage doesn't incur the overhead of creating all of the delegates. It should be reasonably common to hit little to no delegate creation.